src/Core/Util/DateTime/DateTime.php

```diff
@@ -46,8 +46,18 @@ final class DateTime
     /**
      * DateTime value which should be considered same as null
      */
+    public const NULL_DATETIME = '0000-00-00 00:00:00';
+
+    /**
+     * @deprecated use NULL_DATETIME or NULL_DATE depending on usecase
+     */
     public const NULL_VALUE = '0000-00-00 00:00:00';
```

Comment:
Never seen this const before but, is there a reason to have introduced this in the past? Why not use the NULL in the database instead? :thinking: 
ping @matks @jolelievre

Reply:
database is using this format as "null" everywhere, so it was introduced some time ago to avoid changing db structure.
I think one option would indeed be to introduce a BC break and allow NULL value in db instead.

Reply:
@PierreRambaud As @zuk3975 explained we are translating here the constraints of the database. Our work is to migrate the Product page from its hybrid current architecture (half legacy half Symfony) to fully Symfony, and we are supposed to do it while preserving the behavior and the underlying legacy logic/structure.

I'm OK to introduce a nullable column (so we could avoid using this weird value) but not at the same time at migrating 😅  it would make things harde to build, review and test.

Reply:
It's not a BC to allowing NULL value now, it adds something and we can handle both, the NULL and 0000-blalb000 :sweat_smile: 
But I understand, let do it later :)